xray/commands/audit/sca/yarn/yarn.go

```diff
@@ -24,6 +48,28 @@ func BuildDependencyTree() (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps [
 	if errorutils.CheckError(err) != nil {
 		return
 	}
+
+	projectInstalled, err := isYarnProjectInstalled(currentDir)
+	if err != nil {
+		return
+	}
+
+	if !projectInstalled || len(params.InstallCommandArgs()) != 0 {
+		// In case project is not "installed" or in case the user has provided an 'install' command to run
+		var serverDetails *config.ServerDetails
+		serverDetails, err = params.ServerDetails()
+		if err != nil {
+			err = fmt.Errorf("failed to get server details while building yarn dependency tree: %s", err.Error())
+			return
+		}
+		depsRepo := params.DepsRepo()
+		err = configureYarnResolutionServerAndRunInstall(params, currentDir, executablePath, serverDetails, depsRepo)
```

Comment:
depsRepo and serverDetails exist in the `AuditParams`, so you don't need to pass them as parameters.